DataFormats/Common/interface/getThinned_implementation.h

```diff
@@ -133,36 +203,235 @@ namespace edm {
         if (!hasAny) {
           continue;
         }
-        // Get the thinned container and set the pointers and indexes into
-        // it (if we can find it)
-        ProductID thinnedCollectionPID = thinnedAssociation->thinnedCollectionID();
-        WrapperBase const* thinnedCollection = getByProductID(thinnedCollectionPID);
+
+        // if the thinned container is also slimmed, store the
+        // association and the keys to thinned for possible later use
+        // and ignore for now
+        if (associatedBranches->isSlimmed()) {
+          assert(slimmedAssociation == nullptr);
+          slimmedAssociation = thinnedAssociation;
+          slimmedIndexes = std::move(thinnedIndexes);
+          continue;
+        }
+
+        // Set the pointers and indexes into the thinned container (if we can find it)
         if (thinnedCollection == nullptr) {
           // Thinned container is not found, try looking recursively in thinned containers
           // which were made by selecting elements from this thinned container.
-          getThinnedProducts(thinnedCollectionPID,
+          auto slimmed = getThinnedOnlyProducts(thinnedCollectionPID,
```

Comment:
slimmedOrThinned might be a better name instead of slimmed.

Also (probably you already realized this) the comment to getThinnedProduct also applies to this function.

Reply:
Forget this comment. "slimming" is a good name. I didn't see the in the thinning case the output is in the arguments.

Reply:
I'm not very happy with the asymmetry in return types between this and singular `getThinnedOnlyProduct()`, but I haven't managed to find a nice way to return the thinning part via return values that would not lead to "excessive" copying of vector contents (mostly the `std::vector<WrapperBase const*>& foundContainers`).